sdk/tables/Azure.Data.Tables/tests/TableClientLiveTests.cs

```diff
@@ -718,7 +718,7 @@ public async Task UpsertedEntitiesAreRoundtrippedWithProperOdataAnnoations()
                 .ConfigureAwait(false);
 
             Assert.That(entityResults.First()[StringTypePropertyName], Is.TypeOf<string>(), "The entity property should be of type string");
-            Assert.That(entityResults.First()[DateTypePropertyName], Is.TypeOf<DateTimeOffset>(), "The entity property should be of type DateTime");
+            Assert.That(entityResults.First()[DateTypePropertyName], Is.TypeOf<DateTime>(), "The entity property should be of type DateTime");
```

Comment:
Is this a breaking change?

Reply:
I don't think so since DateTime -> DateTimeOffset is an implicit conversion and the changes to GetDateTime and GetDateTimeOffset.

Reply:
Wouldn't the following break?

`var d = (DateTimeOffset)entityResults.First()[DateTypePropertyName]`

Reply:
hmm, yeah you are right. The following is what is required for the implicit cast. 

`DateTimeOffset d = (DateTime)entityResults.First()[DateTypePropertyName]`

I'll revert the type change and leave the changes to `GetDateTime` and `GetDateTimeOffset` which still resolves the specific scenario mentioned in the bug.

Reply:
`var d = (DateTimeOffset)entityResults.First()[DateTypePropertyName]`  -  doesn't this represent an incorrect value today?

Reply:
yes, unfortunately. But at least that would only be an issue for new code and not break existing code that does this.

Reply:
I dunno... I understand the goal, but I would prefer that you break me so that I know to fix my code rather than my silently receiving an incorrect value, personally.

Reply:
I could go either way - @pakrym, do you feel strongly one way or the other?

Reply:
Can we fix `(DateTimeOffset)entityResults.First()[DateTypePropertyName]` so it represents to correct value and add `GetDateTime` as well ?

Reply:
Okay, chatted with @christothes  and understand the problem now.

The DateTimeOffset is the correct type to use as the default to represent dates. Changing it would be a breaking change as well. I think the only change here is fixing the GetDateTime() method.

Reply:
Discussed a bit offline. We'll go with the leaving out the breaking change since the value, even as a DateTimeOffset type is still equivalent from a value perspective.